ec/ecresource/deploymentresource/topology_dedicated_masters.go

```diff
@@ -75,13 +77,21 @@ func UpdateDedicatedMasterTier(
 
 	if nodesInCluster < dedicatedMastersThreshold {
 		// Disable master tier
-		resp.Plan.SetAttribute(ctx,
-			path.Root("elasticsearch").AtName("master").AtName("size"),
-			"0g",
-		)
+		if planElasticsearch.MasterTier.IsUnknown() || planElasticsearch.MasterTier.IsNull() {
+			resp.Plan.SetAttribute(ctx,
+				path.Root("elasticsearch").AtName("master"),
+				types.ObjectNull(es.ElasticsearchTopologyAttrs()),
+			)
+		} else {
+			resp.Plan.SetAttribute(ctx,
+				path.Root("elasticsearch").AtName("master").AtName("size"),
+				"0g",
```

Comment:
Do I get it correctly that  even if `master` tier is defined in plan (because it's, e.g. defined in config), we remove it by setting size to 0? IMO this is the correct behavior.

Reply:
It is changed if it is already in the plan, not if it is already set in the config (There is a check earlier that stops the logic if the config defines the master).

This happens if the master tier had existed before, for example if it was automatically added. Or if it was configured, but the config was removed. In that case we keep whatever config the plan has and only change the size. 

If I just completely remove the whole tier (by setting it to null) I was running into some problems because the update logic couldn't handle a null master tier if a master had existed previously in the deployment.  But I think this is anyway better as it causes a smaller change and we can keep info about currently used instance-configs in the plan.